src/BotGroups.pas

```diff
@@ -0,0 +1,967 @@
+  D3DXMatrixMultiply(matWorld2, mat2, matWorld2);
+
+  if (ownProps.fegyv = FEGYV_HPL) and state.aiming then
+      D3DXMatrixTranslation(matWorld, -0.05, 0.02, -0.11)
+    else
+      if (ownProps.fegyv = FEGYV_M82A1) and not state.aiming then
+        D3DXMatrixTranslation(matWorld, -0.05, -0.05, 0.15)
+      else
+        if (ownProps.fegyv = FEGYV_BM3) and state.aiming then
+          D3DXMatrixTranslation(matWorld, -0.05, -0.05, 0.06)
+        else
+          if (ownProps.fegyv = FEGYV_BM3) and not state.aiming then
+            D3DXMatrixTranslation(matWorld, -0.05, 0.01, -0.08)
+          else
+            if (ownProps.fegyv = FEGYV_LAW) then
+              D3DXMatrixTranslation(matWorld, 0.05, 0, 0)
+            else
+              if (ownProps.fegyv = FEGYV_H31_T)
+              or (ownProps.fegyv = FEGYV_H31_G) then
+                D3DXMatrixTranslation(matWorld, 0.05, 0, 0)
+              else
+                if (ownProps.fegyv = FEGYV_QUAD) and state.aiming then
+                  D3DXMatrixTranslation(matWorld, -0.00, -0.1, -0.03)
+                else
+                  D3DXMatrixTranslation(matWorld, -0.05, 0, 0);
+
+   //if (ppl[mi].pos.state and MSTAT_MASK) = 8 then
+   // D3DXMatrixTranslation(matWorld, -0.2, 0.2, -0.1);
+
+
+  D3DXMatrixMultiply(matWorld2, matWorld, matWorld2);
+
+  if not state.aiming then
+    D3DXMatrixTranslation(matWorld, state.pos.x, state.pos.y + 1.2, state.pos.z)
+  else
+    D3DXMatrixTranslation(matWorld, state.pos.x, state.pos.y + 1.5, state.pos.z);
+
+  D3DXMatrixMultiply(matWorld, matWorld2, matWorld);
+
+  result:=matWorld;
+end;
+
+procedure TBot.resetMozgas;
+begin
+  state.mozgas := D3DXVector3(0, 0, 0);
+end;
+
+procedure TBot.applyMozgas;
+begin
+  D3DXVec3Add(state.pos, state.pos, state.mozgas);
+  if (state.mozgas.x <> 0) or (state.mozgas.z <> 0) then
+    state.lastMozgas := D3DXVector3(state.mozgas.x, 0, state.mozgas.z);
+end;
+
+function TBot.hasOjjektumCollide(): boolean;
+var
+  ojjektumIndex, ojjektumInstanceIndex: Integer;
+  tmpVec: TD3DXVector3;
+  distance: Single;
+label skipEgyet;
+begin
+  result := FALSE;
+  tmpVec := state.mozgas;
+  tmpVec.y := state.mozgas.y + 0.1;
+  if state.mozgas.x <> 0 then
+    tmpVec.x := ceil(state.mozgas.x) + (sin(state.rotateY) * 2);
+  if state.mozgas.z <> 0
+    then tmpVec.z := ceil(state.mozgas.z) + (cos(state.rotateY) * 2);
+
+  D3DXVec3Add(tmpVec, state.pos, tmpVec);
+  for ojjektumIndex := 0 to high(ojjektumnevek) do
+    for ojjektumInstanceIndex := 0 to ojjektumarr[ojjektumIndex].hvszam - 1 do
+    begin
+      distance :=
+        tavPointPoint(state.pos, ojjektumarr[ojjektumIndex].holvannak[ojjektumInstanceIndex]);
+      if distance > 50 then goto skipEgyet;  
+      if ojjektumarr[ojjektumIndex].raytestbol(state.pos, tmpVec, ojjektumInstanceIndex, COLLISION_SOLID) then
+      begin
+        result := TRUE;
+        exit;
+      end;
+      skipEgyet:
+    end;
+end;
+
+//optional TODO: remove code duplication
```

Comment:
Jah, valojaban kene egy ralatokE fv, es itt a findTarget csak valaszt egy random enemyt vagy a jatekost, es fogja a post meg muks gombjeit azt csok. Muks nem is kell igazan.